consensus/core/src/tests/base_committer_tests.rs

```diff
@@ -0,0 +1,732 @@
+        context.clone(),
+        Arc::new(MemStore::new()),
+    )));
+    let committer = BaseCommitterBuilder::new(context.clone(), dag_state.clone()).build();
+
+    let wave_length = DEFAULT_WAVE_LENGTH;
+
+    // Add enough blocks to reach the leader round of wave 1.
+    let leader_round_wave_1 = wave_length;
+    let references_leader_round_wave_1 = build_dag(
+        context.clone(),
+        dag_state.clone(),
+        None,
+        leader_round_wave_1,
+    );
+
+    // Filter out that leader.
+    let leader_wave_1 = committer
+        .elect_leader(leader_round_wave_1)
+        .expect("should have elected leader");
+    let references_without_leader_wave_1: Vec<_> = references_leader_round_wave_1
+        .iter()
+        .cloned()
+        .filter(|x| x.author != leader_wave_1.authority)
+        .collect();
+
+    // Create a dag layer where only one authority votes for the leader of wave 1.
+    let mut authorities = context.committee.authorities();
+    let connections_leader_wave_1 = vec![(
+        authorities.next().unwrap().0,
+        references_leader_round_wave_1,
+    )];
+    let connections_without_leader_wave_1: Vec<_> = authorities
+        .take((context.committee.quorum_threshold() - 1) as usize)
+        .map(|authority| (authority.0, references_without_leader_wave_1.clone()))
+        .collect();
+
+    let connections_voting_round_wave_1 = connections_leader_wave_1
+        .into_iter()
+        .chain(connections_without_leader_wave_1)
+        .collect();
+    let references_voting_round_wave_1 =
+        build_dag_layer(connections_voting_round_wave_1, dag_state.clone());
+
+    // Add enough blocks to reach the decision round of wave 1.
+    let decision_round_wave_1 = (2 * wave_length) - 1;
+    build_dag(
+        context.clone(),
+        dag_state.clone(),
+        Some(references_voting_round_wave_1),
+        decision_round_wave_1,
+    );
+
+    // Ensure we directly mark leader of wave 1 undecided as there are less than
+    // 2f+1 blames and 2f+1 certs
+    info!("Try direct commit for leader {leader_wave_1}");
+    let leader_status = committer.try_direct_decide(leader_wave_1);
+    info!("Leader commit status: {leader_status}");
+
+    if let LeaderStatus::Undecided(undecided_slot) = leader_status {
+        assert_eq!(undecided_slot, leader_wave_1)
+    } else {
+        panic!("Expected an undecided leader")
+    };
+
+    // Ensure we indirectly mark leader of wave 1 undecided as there is no anchor
+    // to make an indirect decision.
+    info!("Try indirect commit for leader {leader_wave_1}");
+    let leader_status = committer.try_indirect_decide(leader_wave_1, [].iter());
+    info!("Leader commit status: {leader_status}");
+
+    if let LeaderStatus::Undecided(undecided_slot) = leader_status {
+        assert_eq!(undecided_slot, leader_wave_1)
+    } else {
+        panic!("Expected an undecided leader")
+    };
+}
+
+#[test]
+fn test_byzantine_validator() {
```

Comment:
Shall we have in comments the definition of what we consider here a byzantine validator?